grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -755,4 +789,199 @@ class OrderService {
+                if (qtyParsed) {
+                    def linesWithSameOrderItem = lineItems.findAll { it.id == line.id || (!it.id && it.productCode == line.productCode) }
+                    def qtyToShipTotal
+                    if (linesWithSameOrderItem.size() > 1) {
+                        linesWithSameOrderItem.each { it ->
+                            try {
+                                it.quantityToShip = Integer.parseInt(it.quantityToShip.toString())
+                            } catch (Exception e) {
+                                valid = false
+                            }
+                        }
+                        qtyToShipTotal = linesWithSameOrderItem.quantityToShip.sum()
+                    } else {
+                        qtyToShipTotal = qtyParsed
+                    }
+                    if (orderItem && qtyToShipTotal > orderItem.quantityRemaining) {
+                        line.errors << "Qty to ship for product ${line.productCode}, order item ${line.id} is greater than qty available to ship."
+                        valid = false
+                    }
+                }
+            }
+        }
+        return valid
+    }
+
+    def saveItemsInShipment(Order order, List importedLines) {
+        Shipment shipment = order.pendingShipment
+        if (!shipment) {
+            shipment = new Shipment()
+            shipment.shipmentNumber = identifierService.generateShipmentIdentifier()
+            shipment.expectedShippingDate = new Date()
+            shipment.name = order.name ?: order.orderNumber
+            shipment.description = order.orderNumber
+            shipment.origin = order.origin
+            shipment.destination = order.destination
+            shipment.createdBy = order.orderedBy
+            shipment.shipmentType = ShipmentType.get(Constants.DEFAULT_SHIPMENT_TYPE_ID)
+            shipment.save()
+        } else {
+            if (shipment.shipmentItems) {
+                shipment.shipmentItems.flatten().toArray().each { ShipmentItem shipmentItem ->
+                    shipmentItem.orderItems?.flatten()?.toArray()?.each { OrderItem orderItem ->
+                        orderItem.removeFromShipmentItems(shipmentItem)
+                        shipmentItem.removeFromOrderItems(orderItem)
+                    }
+                    shipment.removeFromShipmentItems(shipmentItem)
+                    shipmentItem.delete()
+                }
+            }
+        }
+
+        importedLines.each { line ->
+            if (line.quantityToShip > 0) {
+                OrderItem orderItem = order.orderItems.find { it.id == line.id }
+                Product product = line.productCode ? Product.findByProductCode(line.productCode) : orderItem.product
+
+                InventoryItem inventoryItem = null
+                if (line.lotNumber && line.expiration) {
+                    inventoryItem = inventoryService.findOrCreateInventoryItem(product, line.lotNumber, line.expiration)
```

Comment:
If there is no existing inventory item then a new one should be created (It's specified in ticket's description)